clients/vscode-hlasmplugin/syntaxes/hlasm.tmLanguage.json

```diff
@@ -233,92 +170,174 @@
+          "patterns": [
+            {
+              "include": "#additionalAsmHighlight"
+            }
+          ]
+        }
+      },
       "patterns": [
         {
-          "match": "(\\S*?)([LKNT]')(.*)",
-          "captures": {
-            "1": {
-              "name": "keyword.control.hlasm",
-              "patterns": [
-                {
-                  "include": "#additionalAsmHighlight"
-                }
-              ]
+          "include": "#parenthesis"
+        },
+        {
+          "include": "#stringType"
+        }
+      ]
+    },
+    "parenthesis": {
+      "patterns": [
+        {
+          "begin": "[(]\\s*(?<=^.{0,71})",
+          "end": "(?=^\\s*\\S(?<=^.{0,15}))|(?=^.{80}\\s*\\S)|([)](?<=^.{0,71}))|(((?<=^.{71}\\s.*)|(?<=^.{0,71}))$)",
+          "applyEndPatternLast": 0,
+          "patterns": [
+            {
+              "include": "#asmArguments"
             },
-            "2": {
-              "name": "support.type.hlasm"
+            {
+              "include": "#lineEnd"
+            }
+          ]
+        }
+      ]
+    },
+    "stringType": {
+      "patterns": [
+        {
+          "begin": "(')(?<=^.{0,71})",
+          "end": "(?=^\\s*\\S(?<=^.{0,15}))|(?=^.{80}\\s*\\S)|((.*?')(?<=^.{0,71})|([^']*)(?<=^.{71})(?=\\s)|(^.{0,71}$))",
+          "applyEndPatternLast": 0,
+          "beginCaptures": {
+            "1": {
+              "name": "string.hlasm"
+            }
+          },
+          "endCaptures": {
+            "1": {
+              "name": "string.hlasm"
+            }
+          },
+          "patterns": [
+            {
+              "match": ".+(?<=^.{0,71})",
+              "name": "string.hlasm"
             },
-            "3": {
-              "patterns": [
-                {
-                  "include": "#asmArguments"
-                }
-              ]
+            {
+              "include": "#lineEnd"
             }
-          }
+          ]
         }
       ]
     },
-    "labels": {
-      "match": "[*]",
+    "lineEnd": {
+      "patterns": [
+        {
+          "include": "#continuationSymbol"
+        },
+        {
+          "include": "#sequenceNumbers"
+        }
+      ]
+    },
+    "additionalAsmHighlight": {
+      "patterns": [
+        {
+          "include": "#nameField"
+        },
+        {
+          "include": "#asmSpecialStatements"
+        },
+        {
+          "include": "#operators"
+        },
+        {
+          "include": "#numbers"
+        }
+      ]
+    },
+    "nameField": {
+      "patterns": [
+        {
+          "include": "#sequenceSymbolNameField"
+        },
+        {
+          "include": "#label"
+        }
+      ]
+    },
+    "sequenceSymbolNameField": {
+      "match": "^\\.\\S+(?<=^.{0,71})",
+      "name": "keyword.control.hlasm"
+    },
+    "sequenceSymbolOperandField": {
+      "match": "\\.[a-zA-Z@$#_0-9]+(?=\\s|$)(?<=^.{0,71})",
+      "name": "keyword.control.hlasm"
+    },
```

Comment:
should we go for the strict sequence symbol rules? `\\.[A-Za-z@#$_][A-Za-z@#$_0-9]*`

Reply:
Sure!